### PR TITLE
Transaction refactor

### PR DESCRIPTION
**Pull request by nhunzaker**
I'm starting to write the Microcosm dev tools and I noticed a couple of issues with our approach. Specifically, when I send transactions over `window.postMessage` to the Chrome extension, it includes a reference to the action type as a function. This is a problem when the browser attempts to serialize the object using this algorithm:

https://developer.mozilla.org/en-US/docs/Web/Guide/API/DOM/The_structured_clone_algorithm

I still do not have an answer yet for `Error` objects returned by promises. They also don't serialize (which is really unfortunate). Still, this lays the groundwork.

### CODE REVIEW

File: examples/simple-svg/components/Viget.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ export default React.createClass({
   getDefaultProps() {
     return {
       title: 'Microcosm SVG Chart Example',
-      height: 250,
+      height: 300,
       width: 250
     }
   },
```
Comment (nhunzaker) [inline examples/simple-svg/components/Viget.jsx]: Probably shouldn't be in here, but the clipping was bugging me :)

File: src/Store.js [hunk 2]
```diff
@@ -3,35 +3,22 @@
...
 
-    return attempt(Store.register(store), action.toString(), [ state, body ], state, store)
-  }
+exports.send = function (store, state, { payload, type }) {
+  let handler = 'register' in store ? store.register()[type] : false
 
+  return handler ? handler.call(store, state, payload) : state
 }
-
-module.exports = Store
```
Comment (nhunzaker) [inline src/Store.js]: I dropped `attempt`. It was super hard to parse and, while this is verbose, I like the clarity

File: src/Microcosm.js [hunk 3]
```diff
@@ -75,7 +74,7 @@ Microcosm.prototype = {
    */
   clean(transaction) {
     if (this.shouldRejectTransaction(transaction)) {
-      this.transactions = this.transactions.filter(t => t !== transaction)
+      this.transactions.splice(this.transactions.indexOf(transaction), 1)
```
Comment (nhunzaker) [inline src/Microcosm.js]: Small performance thing. This helps quite a bit when displaying high frequency data visualizations on mobile

File: src/plugin.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const defaults = {
   },
 
   register(app, options, next) {
-    next()
+    next(null)
```
Comment (nhunzaker) [inline src/plugin.js]: Just in case someone checks `arguments.length` for errors. Who knows?

**Comment by ltk** [comment]
:+1::+1:
